src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +34,62 @@ public class FindCommandParser implements Parser<FindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS);
+
         String trimmedArgs = args.trim();
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        // Check for the presence of any prefix
+        boolean hasNamePrefix = argMultimap.getValue(PREFIX_NAME).isPresent();
+        boolean hasPhonePrefix = argMultimap.getValue(PREFIX_PHONE).isPresent();
+        boolean hasEmailPrefix = argMultimap.getValue(PREFIX_EMAIL).isPresent();
+        boolean hasAddressPrefix = argMultimap.getValue(PREFIX_ADDRESS).isPresent();
+
+        if (hasNamePrefix) {
+            String nameInput = argMultimap.getValue(PREFIX_NAME).get().trim(); // Get the actual name input
+            if (nameInput.isEmpty()) {
+                throw new ParseException("Name cannot be empty!");
```

Comment:
It might be useful to abstract the Message to a variable and refer to it so next time if there's any changes you can just change that variable. Not urgent so maybe in future iterations

Reply:
Yea can, thanks!